### PR TITLE
[SM-923] Project -> Service Accounts : Using new access selector for Project Service Account Access

### PR DESCRIPTION
**Pull request by cd-bitwarden**
## Type of change

```
- [ ] Bug fix
- [ x ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Converting access policy logic to use the new logic.

## Code changes

**bitwarden_license/bit-web/src/app/secrets-manager/models/view/access-policy.view.ts** - adding ProjectServiceAccountsAccessPoliciesView to access-policy.view.ts

**bitwarden_license/bit-web/src/app/secrets-manager/projects/project/project-people.component.ts** - renaming the import convertTOProjectPeopleAccessPolicyItemViews instead of "convertTOAccessPolicyItemViews"

**bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy-selector/models/ap-item-view.type.ts** - renaming converToAccessPolicyItemViews to convertToProjectPeopleAccessPolciyItemViews. Added "convertToProjectServiceAccountsAccessPolicyItemViews" function to use the ProjectServiceAccountsAccessPoliciesView to create a list of ApItemViewType[]

**bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/models/responses/project-access-policies.response.ts** - simple renames

**bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/models/responses/project-service-accounts-access-policies.response.ts** - added project-service-accounts-access-policies.response to fill out ProjectServiceAccountsAccessPolicyResponse[];

**bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy.service.ts** - getProjectServiceAccountsAccessPolicies - used to call server side to gather ProjectServiceAccoutnsAccessPoliciesView
putProjectServiceAccountsAccessPolicies - used to call server side to put/save the ProjectServiceAccountsAccessPoliciesView
createProjectServiceAccountAccessPolicyView - function used to create the ProjectServiceAccountAccessPoliciesView needed for get/putProjectServiceAccountsAccessPolicies.
getProjectServiceAccountsAccessPoliciesRequest - used to create the request object
## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy-selector/models/ap-item-view.type.ts [hunk 1]
```diff
@@ -63,6 +66,26 @@ export function convertToAccessPolicyItemViews(
   return accessPolicies;
 }
 
+export function convertToProjectServiceAccountsAccessPolicyItemViews(
```
Comment (cd-bitwarden) [inline bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy-selector/models/ap-item-view.type.ts:69]: @Thomas-Avery looking back on this, it seems like I could have just updated it a little and used the convertPotentialGranteesToApItemTypeViewType(), and I shouldn't have renamed the export function from convertToAccessPolicyItemViews to convertToProjectPeopleAccessPolicyItemViews 🤔 what are your thoughts?

Comment (cd-bitwarden) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy-selector/models/ap-item-view.type.ts:69]: Do we want this to be generic, or do we intend on making specific ones for each scenario? I'm having a hard time remembering if we had a convo about this 🤔

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy.service.ts [hunk 2]
```diff
@@ -607,4 +617,78 @@ export class AccessPolicyService {
...
+  ): ProjectServiceAccountsAccessPoliciesView {
+    const view = new ProjectServiceAccountsAccessPoliciesView();
+
+    view.serviceAccountAccessPolicies =
+      projectServiceAccountsAccessPoliciesResponse.serviceAccountAccessPolicies.map((ap) => {
+        return this.createProjectServiceAccountAccessPolicyView(ap);
+      });
+
+    return view;
+  }
```
[the remark this answers is not in the document]
Comment (cd-bitwarden) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy.service.ts:678]: @Thomas-Avery I think I get what you're asking me to do, please double check this just to be sure 😆

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy.service.ts [hunk 3]
```diff
@@ -34,18 +36,22 @@ import {
...
 
 @Injectable({
   providedIn: "root",
 })
 export class AccessPolicyService {
-  private _projectAccessPolicyChanges$ = new Subject<ProjectAccessPoliciesView>();
   private _serviceAccountAccessPolicyChanges$ = new Subject<ServiceAccountAccessPoliciesView>();
+  private _projectAccessPolicyChanges$ = new Subject<ProjectServiceAccountsAccessPoliciesView>();
+  private _projectServiceAccountAccessPolicyChanges$ =
+    new Subject<ProjectServiceAccountsAccessPoliciesView>();
```
[the remark this answers is not in the document]
Comment (cd-bitwarden) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/access-policies/access-policy.service.ts:53]: @Thomas-Avery are you okay with me removing this as a part of the SM-1030? It looks like removing it now properly effects other areas like access-removal-dialog.component.ts and service-account-people.component.ts, and a few others.

**Review by Thomas-Avery — APPROVED** [review]
Looks good

**Comment by Thomas-Avery** [comment]
Closing this work moved to https://github.com/bitwarden/clients/pull/8789

**Pull request closed without merging**